File: browser_test.py
"""Playwright browser smoke test for Organt Chat"""
import json, time
from playwright.sync_api import sync_playwright

def run():
    checks = []
    errors = []

    with sync_playwright() as pw:
        browser = pw.chromium.launch(headless=True)
        ctx     = browser.new_context(viewport={"width": 1280, "height": 800})
        page    = ctx.new_page()

        page.on("console", lambda msg: errors.append(f"[console.{msg.type}] {msg.text}") if msg.type == "error" else None)
        page.on("pageerror", lambda e: errors.append(f"[pageerror] {e}"))

        page.goto("http://localhost:3000", wait_until="networkidle", timeout=12000)

        # 1. App shell
        app = page.query_selector(".app")
        checks.append("app shell: " + ("PASS" if app else "FAIL"))

        # 2. Channel list (wait for WS init)
        page.wait_for_selector(".channel-item", timeout=6000)
        items = page.query_selector_all(".channel-item")
        checks.append(f"channel list: {'PASS' if len(items) >= 3 else 'FAIL'} ({len(items)} items)")

        # 3. Username
        uname = page.text_content("#userNameEl")
        checks.append(f"username: {'PASS' if uname and uname != '연결 중…' else 'FAIL'} ({uname})")

        # 4. Topbar channel
        ch_name = page.text_content("#topbarChannelName")
        checks.append(f"topbar channel: {'PASS' if ch_name and ch_name != '채널을 선택하세요' else 'FAIL'} ({ch_name})")

        # 5. Composer visible
        composer = page.query_selector(".composer__row")
        checks.append("composer: " + ("PASS" if composer else "FAIL"))

        # 6. Send a message
        page.click("#composerInput")
        page.keyboard.type("안녕하세요 브라우저 테스트")
        page.keyboard.press("Enter")
        page.wait_for_selector(".message", timeout=5000)
        msg_text = page.text_content(".message__text") or ""
        checks.append(f"message send: {'PASS' if '브라우저 테스트' in msg_text else 'FAIL'} ({msg_text[:40]})")

        # 7. Tone badge (AI disabled → uncertain after analysis)
        try:
            page.wait_for_selector(".message__tone", timeout=5000)
            grade = page.get_attribute(".message__tone", "data-grade")
            valid = grade in ("positive","neutral","negative","uncertain")
            checks.append(f"tone badge: {'PASS' if valid else 'FAIL'} ({grade})")
        except:
            checks.append("tone badge: SKIP (timeout - AI key needed for tone)")

        # 8. Add channel button
        add_btn = page.query_selector("#addChannelBtn")
        checks.append("add channel btn: " + ("PASS" if add_btn else "FAIL"))

        # 9. Summary button
        sb = page.query_selector("#summaryBtn")
        checks.append("summary btn: " + ("PASS" if sb else "FAIL"))

        # 10. Dark mode (body bg not white)
        bg = page.evaluate("() => getComputedStyle(document.body).backgroundColor")
        checks.append(f"dark bg: {'PASS' if bg and bg != 'rgba(0, 0, 0, 0)' else 'FAIL'} ({bg})")

        # 11. Desktop hamburger hidden
        ham_visible = page.is_visible("#hamburgerBtn")
        checks.append(f"desktop: hamburger hidden: {'PASS' if not ham_visible else 'INFO'}")

        # 12. Mobile: hamburger visible
        page.set_viewport_size({"width": 375, "height": 812})
        time.sleep(0.3)
        ham_mobile = page.is_visible("#hamburgerBtn")
        checks.append(f"mobile: hamburger visible: {'PASS' if ham_mobile else 'FAIL'}")

        # 13. Mobile sidebar toggle
        page.click("#hamburgerBtn")
        time.sleep(0.3)
        sidebar_open = page.query_selector(".sidebar--open") is not None
        checks.append(f"mobile sidebar toggle: {'PASS' if sidebar_open else 'FAIL'}")

        # 14. Screenshot (desktop)
        page.set_viewport_size({"width": 1280, "height": 800})
        page.screenshot(path="/home/user/organt_workspace/p-024-채팅-사이트-with-ai/screenshot.png")
        checks.append("screenshot: PASS")

        browser.close()

    print("=== BROWSER CHECKS ===")
    for c in checks:
        print(c)

    if errors:
        print("=== JS ERRORS ===")
        for e in errors:
            print(e)
    else:
        print("=== NO JS ERRORS ===")

if __name__ == "__main__":
    run()

File: browser-test.js
// Playwright browser verification
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page    = await browser.newPage();
  const errors  = [];
  const checks  = [];

  page.on('console', msg => {
    if (msg.type() === 'error') errors.push('[console.error] ' + msg.text());
  });
  page.on('pageerror', err => errors.push('[page.error] ' + err.message));

  await page.goto('http://localhost:3000', { waitUntil: 'networkidle', timeout: 10000 });

  // 1. App shell renders
  const appEl = await page.$('.app');
  checks.push('app shell: ' + (appEl ? 'PASS' : 'FAIL'));

  // 2. Sidebar with channels
  await page.waitForSelector('.channel-item', { timeout: 5000 });
  const channelItems = await page.$$('.channel-item');
  checks.push('channel list: ' + (channelItems.length >= 3 ? 'PASS (' + channelItems.length + ' items)' : 'FAIL'));

  // 3. Username assigned
  const userName = await page.$eval('#userNameEl', el => el.textContent);
  checks.push('username: ' + (userName && userName !== '연결 중…' ? 'PASS (' + userName + ')' : 'FAIL'));

  // 4. Topbar shows channel name
  await page.waitForSelector('.topbar__channel-name', { timeout: 3000 });
  const chName = await page.$eval('#topbarChannelName', el => el.textContent);
  checks.push('topbar channel: ' + (chName && chName !== '채널을 선택하세요' ? 'PASS (' + chName + ')' : 'FAIL'));

  // 5. Composer visible
  const composerEl = await page.$('.composer__row');
  checks.push('composer: ' + (composerEl ? 'PASS' : 'FAIL'));

  // 6. Send a message and verify it appears
  await page.click('#composerInput');
  await page.keyboard.type('안녕하세요 브라우저 테스트입니다');
  await page.keyboard.press('Enter');
  await page.waitForSelector('.message:not(.message--consecutive)', { timeout: 5000 });
  const msgText = await page.$eval('.message__text', el => el.textContent.trim());
  checks.push('message send: ' + (msgText.includes('브라우저 테스트') ? 'PASS' : 'FAIL (' + msgText + ')'));

  // 7. Tone badge appears (async, wait up to 4s)
  try {
    await page.waitForSelector('.message__tone', { timeout: 4000 });
    const grade = await page.$eval('.message__tone', el => el.dataset.grade);
    const validGrades = ['positive','neutral','negative','uncertain'];
    checks.push('tone badge: ' + (validGrades.includes(grade) ? 'PASS (' + grade + ')' : 'FAIL (' + grade + ')'));
  } catch {
    checks.push('tone badge: SKIP (AI disabled — tone=uncertain default applied in 4s)');
  }

  // 8. Summary button visible
  const summaryBtnEl = await page.$('#summaryBtn');
  checks.push('summary btn: ' + (summaryBtnEl ? 'PASS' : 'FAIL'));

  // 9. Mobile: check hamburger hidden on desktop
  const hamburgerVisible = await page.isVisible('#hamburgerBtn');
  checks.push('hamburger (desktop hidden): ' + (!hamburgerVisible ? 'PASS' : 'INFO (check mobile)'));

  // 10. Switch to mobile viewport and check sidebar toggle
  await page.setViewportSize({ width: 375, height: 812 });
  const hamburgerMobile = await page.isVisible('#hamburgerBtn');
  checks.push('hamburger (mobile visible): ' + (hamburgerMobile ? 'PASS' : 'FAIL'));

  // 11. Dark mode default (body background)
  const bgColor = await page.$eval('body', el => getComputedStyle(el).backgroundColor);
  checks.push('dark bg applied: ' + (bgColor !== 'rgba(0, 0, 0, 0)' ? 'PASS (' + bgColor + ')' : 'FAIL'));

  // 12. No JS color setting (grep check done separately)

  // 13. Take screenshot
  await page.setViewportSize({ width: 1280, height: 800 });
  await page.screenshot({ path: '/home/user/organt_workspace/p-024-채팅-사이트-with-ai/screenshot.png', fullPage: false });
  checks.push('screenshot: PASS (saved screenshot.png)');

  await browser.close();

  console.log('=== BROWSER CHECKS ===');
  console.log(checks.join('\n'));
  if (errors.length) {
    console.log('=== ERRORS ===');
    console.log(errors.join('\n'));
  } else {
    console.log('=== NO CONSOLE ERRORS ===');
  }
})().catch(err => { console.error('Test failed:', err.message); process.exit(1); });
